internal/app/osd/internal/reg/containerd.go

```diff
@@ -8,60 +8,81 @@ import (
 	"context"
 	"log"
 	"path"
+	"path/filepath"
 	"strings"
 
 	"github.com/containerd/containerd"
-	"github.com/containerd/containerd/defaults"
 	"github.com/containerd/containerd/namespaces"
+	"github.com/talos-systems/talos/internal/pkg/constants"
 )
 
-type containerProc struct {
-	Name   string // Friendly name
-	ID     string // container sha/id
-	Digest string // Container Digest
-	Status string // Running state of container
-	Pid    uint32
+type container struct {
+	Display string // Friendly Name
+	Name    string // container name
+	ID      string // container sha/id
+	Digest  string // Container Digest
+	Image   string
+	Status  string // Running state of container
+	Pid     uint32
+	LogFile string
+}
+
+type pod struct {
+	Name    string
+	Sandbox string
 
-	Container containerd.Container
-	Context   context.Context
+	Containers []*container
 }
 
 func connect(namespace string) (*containerd.Client, context.Context, error) {
-	client, err := containerd.New(defaults.DefaultAddress)
+	client, err := containerd.New(constants.ContainerdAddress)
 	return client, namespaces.WithNamespace(context.Background(), namespace), err
 }
 
-func containerID(namespace string) ([]containerProc, error) {
+// nolint: gocyclo
+func podInfo(namespace string) ([]*pod, error) {
+	pods := []*pod{}
+
 	client, ctx, err := connect(namespace)
 	if err != nil {
-		return nil, err
+		return pods, err
 	}
 	// nolint: errcheck
 	defer client.Close()
 
+	var imageList map[string]string
+	imageList, err = images(namespace)
+	if err != nil {
+		return pods, err
+	}
+
 	containers, err := client.Containers(ctx)
 	if err != nil {
-		return nil, err
+		return pods, err
 	}
 
-	cps := make([]containerProc, len(containers))
+	for _, cntr := range containers {
+		cp := &container{}
 
-	for _, container := range containers {
-		cp := containerProc{}
+		info, err := cntr.Info(ctx)
+		if err != nil {
+			log.Println(err)
```

Comment:
I think at least some context would benefit a lot when debugging issues: `log.Printf("error getting container info: %s", err)`

Another option might be to use multierror and collect all the errors for the containers we skipped on the way and return this error with the list of the pods back to the client, so that both could be printed?